pysph/base/opencl.py

```diff
@@ -48,6 +52,91 @@ def get_elwise_kernel(kernel_name, args, src, preamble=""):
     return profile_kernel(knl, kernel_name)
 
 
+def get_simple_kernel(kernel_name, args, src, wgs, preamble=""):
+    ctx = get_context()
+    knl = SimpleKernel(
+        ctx, args, src, wgs,
+        kernel_name, preamble=preamble
+    )
+
+    return profile_kernel(knl, kernel_name)
+
+
+class SimpleKernel(object):
```

Comment:
Should this be contributed upstream to pyopencl?

Reply:
IMO it's very specific to our use case. It doesn't offer much beyond just using a raw kernel. The only reason I'm using it here is so that I can keep using the gpu helpers that load elementwise kernels from mako files.